### PR TITLE
Separate experiment API from experiment table.

### PR DESCRIPTION
**Pull request by pbadillatorrealba**
This pr has as main focus to separate the experiments APIs (CRUD) from the experiments table component. Changes: 

- Moved the experiments API to its own module in `src/api/experiments`. Is by default typed.

- ExperimentsTable component now imports these methods and executes them in an asynchronous wrapper inside the component.

- Interactivity is added through snackbars/toast and loading animations. These can be seen when the getExperiments request fails and when an experiment is deleted (correctly or incorrectly).

- Typescript is enabled on the front end.

- Interfaces are declared for the main application objects: Experiment, Dataset and Run. These can be found in the `src/types` folder.

If this change is approved, it is expected that the rest of the APIs will be declared in the same or similar way as this one; that is, a module with the API declaration and the component imports the functions that it requires.

Note: this pr depends on #36

### CODE REVIEW

File: DashAI/front/src/api/experiment.ts
Comment (cristian-tamblay) [inline DashAI/front/src/api/experiment.ts:1]: If we are switching to typescript on some files, shouldn't we change all the jsx files to ts? What do you think about mixing javascript and typescript?

File: DashAI/front/src/api/experiment.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+}: {
+  id: string;
+  formData: object;
+}): Promise<object> => {
+  const response = await api.patch(`experiment/${id}`, formData);
+  return response.data;
+};
+
+export const deleteExperiment = async (id: string): Promise<object> => {
+  const response = await api.delete(`experiment/${id}`);
```
Comment (cristian-tamblay) [inline DashAI/front/src/api/experiment.ts]: I think we are missing the path here. We also have a .env defining api paths, maybe we should use them here.
```suggestion
  const response = await api.delete(`/api_v1/experiment/${id}`);
```

File: DashAI/front/src/api/experiment.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+};
+
+export const updateExperiment = async ({
+  id,
+  formData,
+}: {
+  id: string;
+  formData: object;
+}): Promise<object> => {
+  const response = await api.patch(`experiment/${id}`, formData);
```
Comment (cristian-tamblay) [inline DashAI/front/src/api/experiment.ts]:
```suggestion
  const response = await api.patch(`/api_v1/experiment/${id}`, formData);
```

**Review by cristian-tamblay — CHANGES_REQUESTED** [review]

File: DashAI/front/src/api/experiment.ts
Comment (pbadillatorrealba) [reply DashAI/front/src/api/experiment.ts:1]: As far as I understand it is not necessary. Just type the ts files and their dependencies. 
It appears that changes may be gradual.
In this case it was only necessary to type the interfaces and the API file.

**Review by cristian-tamblay — APPROVED** [review]

**Pull request merged**